# Déjame escuchar la música

# Contenido <a id='back'></a>

* [Introducción](#intro)
* [Etapa 1. Descripción de los datos](#data_review)
    * [Conclusiones](#data_review_conclusions)
* [Etapa 2. Preprocesamiento de datos](#data_preprocessing)
    * [2.1 Estilo del encabezado](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusiones](#data_preprocessing_conclusions)
* [Etapa 3. Prueba de hipótesis](#hypothesis)
    * [3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades](#activity)
* [Conclusiones](#end)

## Introducción <a id='intro'></a>
Como analista de datos, tu trabajo consiste en analizar datos para extraer información valiosa y tomar decisiones basadas en ellos. Esto implica diferentes etapas, como la descripción general de los datos, el preprocesamiento y la prueba de hipótesis.

Siempre que investigamos, necesitamos formular hipótesis que después podamos probar. A veces aceptamos estas hipótesis; otras veces, las rechazamos. Para tomar las decisiones correctas, una empresa debe ser capaz de entender si está haciendo las suposiciones correctas.

En este proyecto, compararás las preferencias musicales de las ciudades de Springfield y Shelbyville. Estudiarás datos reales de transmisión de música online para probar la hipótesis a continuación y comparar el comportamiento de los usuarios y las usuarias de estas dos ciudades.

### Objetivo:
Prueba la hipótesis:
1. La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### Etapas
Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.








[Volver a Contenidos](#back)

## Etapa 1. Descripción de los datos <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás `pandas`, así que impórtalo.

In [1]:
# Importar pandas
import pandas as pd

Lee el archivo `music_project_en.csv` de la carpeta `/datasets/` y guárdalo en la variable `df`:

In [2]:
# Leer el archivo y almacenarlo en df
df = pd.read_csv('/datasets/music_project_en.csv')

Muestra las 10 primeras filas de la tabla:

In [3]:
# Obtener las 10 primeras filas de la tabla df
df.head(10)

userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5  Shelbyville  13:09:41     Friday  
6  Springfield  13:00:07  Wednesday  
7  Springfield  20:47:49  Wednesday  
8  Springfield  09:17:40     Friday  
9  Shelbyville  21:20:49  Wednesday

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [4]:
# Obtener la información general sobre nuestros datos
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB


Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: `object`.

Según la documentación:
- `' userID'`: identificador del usuario o la usuaria;
- `'Track'`: título de la canción;
- `'artist'`: nombre del artista;
- `'genre'`: género de la pista;
- `'City'`: ciudad del usuario o la usuaria;
- `'time'`: la hora exacta en la que se reprodujo la canción;
- `'Day'`: día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:
1. Algunos encabezados están en mayúsculas, otros en minúsculas.
2. Hay espacios en algunos encabezados.
3. `Detecta el tercer problema por tu cuenta y descríbelo aquí`. Hay algunos encabezados que no son tan claros, podrían cambiarse por nombres corto pero que den una idea rápida de la información que almacenan usando la regla de "snake_case", por ejemplo: en vez de "userID" podría decir "user_id", en vez de "Track" podría decir "track_name" o en vez de "City" podría decir "city_user".




### Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles: <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`
Se tiene información de cada vez que un usuario de la ciudad de "Springfield" o "Shelbyville" escucha una canción. Toda las columnas de la tabla son cadenas de texto, que en la librería de pandas se definen como tipo "object". Para entender lo que almacena se puede usar el método "info" para tener una idea general de la información que contiene la tabla y para comprender a detalle qué almacena cada columna es necesario revisar la documentación de la tabla para ver la descripción de lo que contiene cada columna.

`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`
Considero que las columnas proporcionadas son suficientes; sin embargo, tal y como está no permitirá concluir si la hipótesis es correcta o no. Por ello, es necesario procesarla y darle forma a la data, de tal manera que nos ayude a analizarla.
`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`
Sí, la información de la tabla no está procesada, en la información general se observa columnas con datos nulos, los nombres de las columnas necesitan limpiarse ya que tienen espacios vacíos y no están uniformizadas con letras minúsculas. Aún no he validado si hay duplicados pero si lo hubiera habría que eliminarlos ya que distorcionaría la conclusión.

[Volver a Contenidos](#back)

## Etapa 2. Preprocesamiento de datos <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.


### Estilo del encabezado <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [5]:
# Muestra los nombres de las columnas
df.columns

Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')

Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

In [6]:
# Bucle en los encabezados poniendo todo en minúsculas
# definimos una lista que almacenará las columnas con los carácteres en minúsculas "column_lowered"
column_lowered = []

#Usamos el bucle "for" y para cada elemento de la lista pasamos a minúsculas usando el método "lower".
#Y para agregarlo a la lista usamos "append"
for column in df.columns:
    column_lowered.append(column.lower())

df.columns = column_lowered
    
print(df.columns)

Index(['  userid', 'track', 'artist', 'genre', '  city  ', 'time', 'day'], dtype='object')


Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

In [7]:
# Bucle en los encabezados eliminando los espacios
# definimos una lista que almacenará las columnas con los carácteres sin espacios al inicio y al final "column_stripped"
column_stripped = []

#Usamos el bucle "for" y para cada elemento eliminamos los espacios al inicio y final usando el método strip

for column in df.columns:
    column_stripped.append(column.strip())

df.columns = column_stripped
    
print(df.columns)

Index(['userid', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [8]:
# Cambiar el nombre de la columna "userid"
df.rename(columns = {'userid': 'user_id'}, inplace = True)

Comprueba el resultado. Muestra los encabezados una vez más:

In [9]:
# Comprobar el resultado: la lista de encabezados
print(df.columns)

Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


[Volver a Contenidos](#back)

### Valores ausentes <a id='missing_values'></a>
 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [10]:
# Calcular el número de valores ausentes
df.isna().sum()

user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64

No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [11]:
# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'
columns_to_modify = ['track', 'artist', 'genre']

for col in columns_to_modify:
    df[col].fillna('unknown', inplace = True)   

Ahora comprueba el resultado para asegurarte de que después del reemplazo no haya valores ausentes en el conjunto de datos. Para hacer esto, cuenta los valores ausentes nuevamente.

In [12]:
# Contar valores ausentes
df.isna().sum()

user_id    0
track      0
artist     0
genre      0
city       0
time       0
day        0
dtype: int64

[Volver a Contenidos](#back)

### Duplicados <a id='duplicates'></a>
Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [13]:
# Contar duplicados explícitos
df.duplicated().sum()

3826

Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [15]:
# Eliminar duplicados explícitos
df.drop_duplicates(inplace = True)

Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [16]:
# Comprobar de nuevo si hay duplicados

df.duplicated().sum()

0

Ahora queremos deshacernos de los duplicados implícitos en la columna `genre`. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:
* Extrae la columna `genre` del DataFrame.
* Llama al método que devolverá todos los valores únicos en la columna extraída.


In [17]:
# Inspeccionar los nombres de géneros únicos
df['genre'].unique()

array(['rock', 'pop', 'folk', 'dance', 'rusrap', 'ruspop', 'world',
       'electronic', 'unknown', 'alternative', 'children', 'rnb', 'hip',
       'jazz', 'postrock', 'latin', 'classical', 'metal', 'reggae',
       'triphop', 'blues', 'instrumental', 'rusrock', 'dnb', 'türk',
       'post', 'country', 'psychedelic', 'conjazz', 'indie',
       'posthardcore', 'local', 'avantgarde', 'punk', 'videogame',
       'techno', 'house', 'christmas', 'melodic', 'caucasian',
       'reggaeton', 'soundtrack', 'singer', 'ska', 'salsa', 'ambient',
       'film', 'western', 'rap', 'beats', "hard'n'heavy", 'progmetal',
       'minimal', 'tropical', 'contemporary', 'new', 'soul', 'holiday',
       'german', 'jpop', 'spiritual', 'urban', 'gospel', 'nujazz',
       'folkmetal', 'trance', 'miscellaneous', 'anime', 'hardcore',
       'progressive', 'korean', 'numetal', 'vocal', 'estrada', 'tango',
       'loungeelectronic', 'classicmetal', 'dubstep', 'club', 'deep',
       'southern', 'black', 'folkrock', 

Busca en la lista para encontrar duplicados implícitos del género `hiphop`. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para deshacerte de ellos, crea una función llamada `replace_wrong_genres()` con dos parámetros:
* `wrong_genres=`: esta es una lista que contiene todos los valores que necesitas reemplazar.
* `correct_genre=`: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna `'genre'` de la tabla `df`, es decir, remplazar cada valor de la lista `wrong_genres` por el valor en `correct_genre`.

Dentro del cuerpo de la función, utiliza un bucle `'for'` para iterar sobre la lista de géneros incorrectos, extrae la columna `'genre'` y aplica el método `replace` para hacer correcciones.

In [18]:
# Función para reemplazar duplicados implícitos
def replace_wrong_genres(wrong_genres, correct_genre):
    for wrong_genre in wrong_genres:
        df['genre'].replace(wrong_genre, correct_genre, inplace = True)


Ahora, llama a `replace_wrong_genres()` y pásale tales argumentos para que retire los duplicados implícitos (`hip`, `hop` y `hip-hop`) y los reemplace por `hiphop`:

In [19]:
# Eliminar duplicados implícitos
replace_wrong_genres(['hip','hop','hip-hop'], 'hiphop')

Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna `'genre'` una vez más:

In [20]:
# Comprobación de duplicados implícitos
df['genre'].unique()

array(['rock', 'pop', 'folk', 'dance', 'rusrap', 'ruspop', 'world',
       'electronic', 'unknown', 'alternative', 'children', 'rnb',
       'hiphop', 'jazz', 'postrock', 'latin', 'classical', 'metal',
       'reggae', 'triphop', 'blues', 'instrumental', 'rusrock', 'dnb',
       'türk', 'post', 'country', 'psychedelic', 'conjazz', 'indie',
       'posthardcore', 'local', 'avantgarde', 'punk', 'videogame',
       'techno', 'house', 'christmas', 'melodic', 'caucasian',
       'reggaeton', 'soundtrack', 'singer', 'ska', 'salsa', 'ambient',
       'film', 'western', 'rap', 'beats', "hard'n'heavy", 'progmetal',
       'minimal', 'tropical', 'contemporary', 'new', 'soul', 'holiday',
       'german', 'jpop', 'spiritual', 'urban', 'gospel', 'nujazz',
       'folkmetal', 'trance', 'miscellaneous', 'anime', 'hardcore',
       'progressive', 'korean', 'numetal', 'vocal', 'estrada', 'tango',
       'loungeelectronic', 'classicmetal', 'dubstep', 'club', 'deep',
       'southern', 'black', 'folkrock

[Volver a Contenidos](#back)

### Tus observaciones <a id='data_preprocessing_conclusions'></a>

`Describe brevemente lo que has notado al analizar duplicados, cómo abordaste sus eliminaciones y qué resultados obtuviste.`
La validación inicial de duplicados dió un gran número de duplicados 3,826 de 65,079 (6% de la data podría habernos conllevado a un análisis erróneo). Para limpiar la información de duplicados se realizó dos fases, la primero consistió en eliminar los duplicados explícitos, los cuales fueron eliminados con el método "drop_duplicates()". La segunda fase consistió en eliminar los duplicados implícitos; es decir, aquellos que no son identificados con el método mencionado previamente sino que se tiene que corregir a traves de funciones dado el volumen de data, aquí nos apoyamos de replace. Luego de aplicar estas dos fases obtuvimos una tabla limpia de duplicados lo cual permitirá iniciar el análisis y evitar conclusiones erradas.

[Volver a Contenidos](#back)

## Etapa 3. Prueba de hipótesis <a id='hypothesis'></a>

### Hipótesis: comparar el comportamiento del usuario o la usuaria en las dos ciudades <a id='activity'></a>

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

* Agrupa a los usuarios y las usuarias por ciudad.
* Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final:
`df.groupby(by='....')['column'].method()`Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.



In [25]:
# Contar las canciones reproducidas en cada ciudad
print(df.groupby(by='city')['day'].count())

city
Shelbyville    18512
Springfield    42741
Name: day, dtype: int64


`Comenta tus observaciones aquí`
Sin filtrar el día de la semana en que el usuario reproduce música se podría afirmar que los ciudadanos de "Springdield" tienen mayor actividad reproduciendo música online que los de la ciudad de "Shelbyville" donde la proporción de reproducciones por ciudad es de 70%-Shelbyville versus 30%-Springfield.

Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.


In [37]:
# Calcular las canciones reproducidas en cada uno de los tres días
#print(df['day'].unique())
print('Número de reproducciones por ciudad los días LUNES')
print(df[df['day'] == 'Monday'].groupby(by='city')['day'].count())
print('\n' + 'Número de reproducciones por ciudad los días MIERCOLES')
print(df[df['day'] == 'Wednesday'].groupby(by='city')['day'].count())
print('\n' + 'Número de reproducciones por ciudad los días VIERNES')
print(df[df['day'] == 'Friday'].groupby(by='city')['day'].count())

Número de reproducciones por ciudad los días LUNES
city
Shelbyville     5614
Springfield    15740
Name: day, dtype: int64

Número de reproducciones por ciudad los días MIERCOLES
city
Shelbyville     7003
Springfield    11056
Name: day, dtype: int64

Número de reproducciones por ciudad los días VIERNES
city
Shelbyville     5895
Springfield    15945
Name: day, dtype: int64


`Comenta tus observaciones aquí`
Los resultados considerando el día y la ciudad nos indican que la actividad por día en cada ciudad varía; sin embargo, se mantiene la tendencia de mayor actividad de reproducciones de los ciudadanos de "Springfield" que los de "Shelville" cada día.

Si vemos el número de reproducciones por ciudad y día se observa que los ciudadanos de "Shelbyville" tienen mayor actividad los días miércoles. Por otro lado los ciudadanos de la ciudad de "Springville" tienen mayor actividad los días lunes y viernes.

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función `number_tracks()` para calcular el número de canciones reproducidas en un determinado día **y** ciudad. La función debe aceptar dos parámetros:

- `day`: un día de la semana para filtrar. Por ejemplo, `'Monday'` (lunes).
- `city`: una ciudad para filtrar. Por ejemplo, `'Springfield'`.

Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [70]:
# Declara la función number_tracks() con dos parámetros: day= y city=.

def number_tracks(day, city):

 # Almacena las filas del DataFrame donde el valor en la columna 'day' es igual al parámetro day=
 data_filtered = df[df['day'] == day]

 # Filtra las filas donde el valor en la columna 'city' es igual al parámetro city=
 city_filtered = data_filtered[data_filtered['city'] == city]

 # Extrae la columna 'user_id' de la tabla filtrada y aplica el método count()
 number_datarows = city_filtered['user_id'].count()

 # Devolve el número de valores de la columna 'user_id'
 return number_datarows

Llama a `number_tracks()` seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [71]:
# El número de canciones reproducidas en Springfield el lunes
day = 'Monday'
city = 'Springfield'
print('On ' + day + ' people from '+ city + ' listened ' + str(number_tracks(day, city)) + ' songs.')

On Monday people from Springfield listened 15740 songs.


In [72]:
# El número de canciones reproducidas en Shelbyville el lunes
day = 'Monday'
city = 'Shelbyville'
print('On ' + day + ' people from '+ city + ' listened ' + str(number_tracks(day, city)) + ' songs.')

On Monday people from Shelbyville listened 5614 songs.


In [73]:
# El número de canciones reproducidas en Springfield el miércoles
day = 'Wednesday'
city = 'Springfield'
print('On ' + day + ' people from '+ city + ' listened ' + str(number_tracks(day, city)) + ' songs.')

On Wednesday people from Springfield listened 11056 songs.


In [74]:
# El número de canciones reproducidas en Shelbyville el miércoles
day = 'Wednesday'
city = 'Shelbyville'
print('On ' + day + ' people from '+ city + ' listened ' + str(number_tracks(day, city)) + ' songs.')

On Wednesday people from Shelbyville listened 7003 songs.


In [75]:
# El número de canciones reproducidas en Springfield el viernes
day = 'Friday'
city = 'Springfield'
print('On ' + day + ' people from '+ city + ' listened ' + str(number_tracks(day, city)) + ' songs.')

On Friday people from Springfield listened 15945 songs.


In [76]:
# El número de canciones reproducidas en Shelbyville el viernes
day = 'Friday'
city = 'Shelbyville'
print('On ' + day + ' people from '+ city + ' listened ' + str(number_tracks(day, city)) + ' songs.')

On Friday people from Shelbyville listened 5895 songs.


**Conclusiones**

`Comenta si la hipótesis es correcta o se debe rechazar. Explica tu razonamiento.`

La hipótesis del proyecto es "La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.".

Luego de haber analizado la información considero que se debe aceptar la hipótesis debido a lo siguiente:

1 Si filtramos solo por ciudad, se observa que el 70% del total de reproducciones fueron de ciudadanos de "Springfield" mientras que el 30% restante fue de "Shellbyville". Por lo que se concluye que los usuarios y usuarias de "Springfield" registraron mayor actividad que los de "Shelbyville" y eso demuestra que existe una diferencia al realizar el análisis considerando la ciudad.

2 Si además del filtro de ciudad consideramos la variable día en el análisis, se observa lo siguiente:

 a) Ciudadanos de Springfield: El 37% del total de reproducciones de usuarios y usuarias de esa ciudad se dieron los lunes, el 26% se dieron los miércoles y el 37% restante se dieron los viernes. Por lo cual se puede concluir que los días lunes y viernes son los días con mayor actividad de reproducciones para los ciudadanos de "Springfield".
 
 b) Ciudadanos de Shellbyville: El 30% del total de reproducciones de usuarios y usuarias de esa ciudad se dieron los lunes, el 38% se dieron los miércoles y el 32% restante se dieron los viernes. Por lo cual se puede concluir que el día miércoles es el día con mayor actividad de reproducciones para los ciudadanos de "Shellbyville".
 
3 Dado que dependiendo si vemos el número de reproducciones por ciudad y día la actividad de los usuarios y usuarias difieren considero que la hipótesis es correcta.


[Volver a Contenidos](#back)

# Conclusiones <a id='end'></a>

Según lo comentado previamente se concluye que la hipótesis es correcta. Y se puede resumir, en base al análisis realizado, lo siguiente:

1 Los usuarios y usuarias de "Springfield" registraron mayor actividad que los de "Shelbyville".

2 Los días lunes y viernes son los días con mayor actividad de reproducciones para los ciudadanos de "Springfield".

3 El día miércoles es el día con mayor actividad de reproducciones para los ciudadanos de "Shellbyville".

[Volver a Contenidos](#back)